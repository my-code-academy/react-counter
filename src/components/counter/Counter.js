import React, { Component } from 'react';
import './counter.css';
import Button from '../Button/Button';

class Counter extends Component {
  counterVal = 0;
  state = {
    count: 0,
  }
  incrementCount = () => {
    this.setState({
      count: this.state.count + 1
    });
  }; 
  decrementCount = () => {
    this.setState({
      count: this.state.count - 1
    });
  }; 
  render() {
    return (
      <div className='counter'>
        <p>Counter: {this.state.count} </p><br/>
        <Button caption='add' onClick={this.incrementCount}></Button>  
        <Button caption='minus' onClick={this.decrementCount}></Button>
      </div>
    );
  }
}

export default Counter;