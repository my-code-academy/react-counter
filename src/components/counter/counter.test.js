import React from 'react';
import renderer from 'react-test-renderer';
import Counter from './Counter';
import {shallow} from 'enzyme';


describe('counter', () => {
  it("should render a counter", () => {
    const tree = renderer.create(<counter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should increment count value', () => {
    const wrapper = shallow(<Counter/>);
    wrapper.instance().incrementCount();
    expect(wrapper.instance().state.count).toEqual(1);
  });

  it('should decrement count value', () => {
    const wrapper = shallow(<Counter/>);
    wrapper.instance().decrementCount();
    expect(wrapper.instance().state.count).toEqual(-1);
  });
});