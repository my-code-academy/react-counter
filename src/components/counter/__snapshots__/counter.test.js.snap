// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter should render a counter 1`] = `<counter />`;
